### PR TITLE
Add WindowFn#assignsToOneWindow

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowFn.java [hunk 1]
```diff
@@ -180,6 +180,13 @@ public boolean isNonMerging() {
   }
 
   /**
+   * Returns true if this {@link WindowFn} always assigns an element to exactly one window.
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowFn.java]: `<p>If it varies per element, or cannot be determined, you should conservatively return false`

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/SlidingWindowsTest.java [hunk 2]
```diff
@@ -55,11 +56,12 @@ public void testSimple() throws Exception {
     expected.put(new IntervalWindow(new Instant(0), new Instant(10)), set(1, 2, 5, 9));
     expected.put(new IntervalWindow(new Instant(5), new Instant(15)), set(5, 9, 10, 11));
     expected.put(new IntervalWindow(new Instant(10), new Instant(20)), set(10, 11));
+    SlidingWindows windowFn = SlidingWindows.of(new Duration(10)).every(new Duration(5));
     assertEquals(
         expected,
-        runWindowFn(
-            SlidingWindows.of(new Duration(10)).every(new Duration(5)),
+        runWindowFn(windowFn,
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/SlidingWindowsTest.java]: Geez, this method makes the test unreadable.

**Comment by kennknowles** [comment]
And LGTM modulo comment comment.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/SlidingWindowsTest.java (hunk 2 above)
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/SlidingWindowsTest.java]: I agree.

https://issues.apache.org/jira/browse/BEAM-2525

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowFn.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowFn.java]: Done.

**Pull request closed without merging**